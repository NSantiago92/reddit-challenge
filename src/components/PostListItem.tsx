import React from "react";
import { Image, StyleSheet, Text, TouchableOpacity, View } from "react-native";
import { Post } from "../model";
import { SubredditNavProps } from "../screens/SubredditScreen";
import kFormatter from "../utils/kFormatter";
import relativeDate from "../utils/relativeDate";

interface PostListItemProps {
  post: Post;
  navigation: SubredditNavProps["navigation"];
}

const PostListItem = ({ post, navigation }: PostListItemProps): JSX.Element => {
  return (
    <TouchableOpacity
      onPress={() => navigation.push("Post", { url: post.url })}
    >
      <View style={styles.container}>
        <View style={styles.scoreContainer}>
          <Text style={styles.score}>{kFormatter(post.score)}</Text>
        </View>
        <View style={styles.detailsContainer}>
          <Text style={styles.title}>{post.title}</Text>
          <View style={styles.detailsSubContainer}>
            <Text style={styles.comments}>{post.numComments} comments</Text>
            <Text style={styles.date}>
              {relativeDate(post.created)} days ago
            </Text>
          </View>
        </View>
        <View style={styles.thumbnailContainer}>
          <Image source={{ uri: post.thumbnail }} style={styles.thumbnail} />
        </View>
      </View>
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  container: {
    flexDirection: "row",
    alignItems: "center",
    padding: 2,
  },
  scoreContainer: {
    width: 45,
    margin: 3,
  },
  score: {
    fontSize: 16,
    textAlign: "center",
    fontWeight: "bold",
  },
  detailsSubContainer: {
    marginTop: 3,
    flexDirection: "row",
    justifyContent: "space-between",
  },
  detailsContainer: {
    flex: 2,
    margin: 5,
  },
  title: {
    fontSize: 13,
    fontWeight: "bold",
  },
  comments: {
    fontSize: 13,
  },
  date: {
    fontSize: 13,
    fontStyle: "italic",
  },
  thumbnailContainer: {},
  thumbnail: {
    width: 64,
    height: 64,
  },
});

export default PostListItem;
