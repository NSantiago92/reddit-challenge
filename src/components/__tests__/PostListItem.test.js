import renderer from "react-test-renderer";
import PostListItem from "../PostListItem";

const testPost = {
  author: "alexflemingart",
  created: new Date(1232850357.0 * 1000),
  id: "vcsgpj",
  numComments: 224,
  score: 4720,
  subreddit: "r/pics",
  thumbnail:
    "https://b.thumbs.redditmedia.com/8qowQW91JVgb5ktjKg6biACUVQ13i5N882Zn0JmPvgY.jpg",
  title:
    "[OC] Here's an African Elephant I drew in charcoal. ~100 hours work. Have a great Wednesday!",
  url: "https://i.redd.it/uvl9v7s9ur591.jpg",
};

it("<PostListItem/> renders correctly", () => {
  const tree = renderer
    .create(<PostListItem post={testPost} goToPost={() => {}} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
