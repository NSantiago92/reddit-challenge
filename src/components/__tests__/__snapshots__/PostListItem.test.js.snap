// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostListItem/> renders correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "padding": 2,
      }
    }
  >
    <View
      style={
        Object {
          "margin": 3,
          "width": 45,
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 16,
            "fontWeight": "bold",
            "textAlign": "center",
          }
        }
      >
        4720
      </Text>
    </View>
    <View
      style={
        Object {
          "flex": 2,
          "margin": 5,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#0000cd",
            "fontSize": 15,
            "fontWeight": "bold",
          }
        }
      >
        <Text
          style={
            Object {
              "backgroundColor": "#dcdcdc",
              "color": "black",
              "fontSize": 13,
              "fontWeight": "normal",
            }
          }
        >
           
          alexflemingart
           
        </Text>
        [OC] Here's an African Elephant I drew in charcoal. ~100 hours work. Have a great Wednesday!
        <Text
          style={
            Object {
              "color": "#575757",
              "fontSize": 13,
              "fontWeight": "normal",
            }
          }
        >
           (
          r/pics
          )
        </Text>
      </Text>
      <View
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "space-between",
            "marginTop": 3,
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 13,
            }
          }
        >
          224
           comments
        </Text>
        <Text
          style={
            Object {
              "fontSize": 13,
              "fontStyle": "italic",
            }
          }
        >
          160 months ago
        </Text>
      </View>
    </View>
    <View
      style={Object {}}
    >
      <Image
        source={
          Object {
            "uri": "https://b.thumbs.redditmedia.com/8qowQW91JVgb5ktjKg6biACUVQ13i5N882Zn0JmPvgY.jpg",
          }
        }
        style={
          Object {
            "height": 64,
            "width": 64,
          }
        }
      />
    </View>
  </View>
</View>
`;
